File: Day1-Arrays/6_BestTimeToBuyAndSellStock.java
import java.util.ArrayList;

class Solution{

    public static int maximumProfit(ArrayList<Integer> prices){
        
        int maxProfit = 0, minVal = Integer.MAX_VALUE;
        
        for(int p : prices)
        {
            minVal = Math.min(minVal, p);
            maxProfit = Math.max(maxProfit, p - minVal);
        }
        
        return maxProfit;
    }

}

File: Day1-Arrays/6_BestTimeToBuyAndSellStock.cpp
#include <bits/stdc++.h> 

int maximumProfit(vector<int> &prices){
    
    int maxProfit = 0, minPrice = INT_MAX;
    for(int &p : prices)
    {
        minPrice = min(minPrice, p);
        maxProfit = max(maxProfit, p - minPrice);
    }
    
    return maxProfit;
}